google/cloud/pubsub_v1/subscriber/_protocol/leaser.py

```diff
@@ -197,8 +198,24 @@ def maintain_leases(self) -> None:
                 #       is inactive.
                 assert self._manager.dispatcher is not None
                 ack_id_gen = (ack_id for ack_id in ack_ids)
-                self._manager._send_lease_modacks(ack_id_gen, deadline)
+                _LOGGER.warning("calling send_lease_modacks with: ")
+                expired_ack_ids = self._manager._send_lease_modacks(
+                    ack_id_gen, deadline
+                )
 
+            start_time = time.time()
+            # If exactly once delivery is enabled, we should drop all expired ack_ids from lease management.
+            if self._manager._exactly_once_delivery_enabled():
+                for ack_id in expired_ack_ids:
+                    self._manager.dispatcher.drop(
```

Comment:
You can make a list of DropRequests and then pass the whole list in, instead of a lot of lists with just one obj.